src/main/java/org/apache/maven/shared/dependency/analyzer/ProjectDependencyAnalysis.java

```diff
@@ -44,19 +44,23 @@
 
     private final Set<Artifact> unusedDeclaredArtifacts;
 
+    private final Set<Artifact> testArtifactsWithNonTestScope;
+
     // constructors -----------------------------------------------------------
 
     public ProjectDependencyAnalysis()
     {
-        this( null, null, null );
+        this( null, null, null, null );
     }
 
     public ProjectDependencyAnalysis( Set<Artifact> usedDeclaredArtifacts, Set<Artifact> usedUndeclaredArtifacts,
```

Comment:
adding an argument incompatibly changes the existing API; consider an additional constructor instead